Calendar: Poster visibility improvements

v1ctorsales opened the pull request:
This PR improves the calendar view by fixing poster scaling and preventing items from collapsing or disappearing at certain zoom levels.

Before: (100% Zoom 1366 x 768 screen)
<img width="1363" height="627" alt="image" src="https://github.com/user-attachments/assets/eae13e83-3bf0-4d6f-9ee0-c1c55550982a" />

After: (100% Zoom 1366 x 768 screen)
<img width="1365" height="625" alt="image" src="https://github.com/user-attachments/assets/792ef212-d256-4673-a6b2-c9fd0ccbccab" />

v1ctorsales committed:
fix(calendar): improve poster visibility and responsive scaling

v1ctorsales committed:
style(calendar): normalize indentation to match project formatting

v1ctorsales committed:
style(calendar): normalize indentation and align with project style

v1ctorsales committed:
chore(calendar): remove duplicated aspect-ratio rule from CSS

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Stremio/stremio-web?pullRequest=1036) <br/>All committers have signed the CLA.

kKaskak commented:
i don't think what you did is correct

<img  height="300" alt="Screenshot 2025-10-11 at 21 42 03" src="https://github.com/user-attachments/assets/c3b89dcf-50bf-442b-9210-03d051c40af9" />
<img height="300" alt="Screenshot 2025-10-11 at 21 44 42" src="https://github.com/user-attachments/assets/dd0a59f8-bf72-4a91-9323-e8c52c21e295" />

v1ctorsales committed:
fix(calendar): apply grid-auto-rows 1fr for equal row height

v1ctorsales commented:
> i don't think what you did is correct
> 


@kKaskak i just commited the fix ;)

kKaskak commented:
there are cases in which no posters are visible for some reason please elaborate

https://github.com/user-attachments/assets/f2b7692d-139d-475a-ac65-949ab2b553a0

v1ctorsales committed:
fix(calendar): redesign cell layout with rows for desktop

v1ctorsales commented:
Thanks for the feedback @kKaskak
I redesigned the calendar cells using flex-direction: row for desktop and reverting to column layout on smaller screens.
Let me know if any changes are still needed.

kKaskak commented:
> Thanks for the feedback @kKaskak I redesigned the calendar cells using flex-direction: row for desktop and reverting to column layout on smaller screens. Let me know if any changes are still needed.

you will need to preserve the aspect ratio of the posters as well since in some cases they look distorted
also the focused state is now partially broken on the current day (see 2nd screenshot)

<img height="300" alt="Screenshot 2025-10-12 at 2 43 53" src="https://github.com/user-attachments/assets/d2fd075c-943a-4ed7-a6d0-82c622c2723c" />

<img height="300" alt="Screenshot 2025-10-12 at 2 43 59" src="https://github.com/user-attachments/assets/e7b0b031-2938-4a4b-849a-2298717a8db0" />

v1ctorsales committed:
fix(calendar): align day and more indicator inline in narrow desktop viewports

v1ctorsales commented:
@kKaskak 

Updated!
- Banner aspect ratio is now preserved.
- The current day highlight no longer breaks.
- On simulated narrow desktop viewports, the day and the "more" icon now appear side by side.
- Mobile layout remains untouched.

kKaskak commented:
> @kKaskak
> 
> Updated!
> 
> * Banner aspect ratio is now preserved.
> * The current day highlight no longer breaks.
> * On simulated narrow desktop viewports, the day and the "more" icon now appear side by side.
> * Mobile layout remains untouched.

are you sure you did it correctly? now the posters are small in some cases
the idea was to remove the 3 dot indicator altogether and always show posters

<img height="300" alt="Screenshot 2025-10-12 at 14 27 49" src="https://github.com/user-attachments/assets/4a28596f-75a2-45e0-bedb-2d1aaee9efac" />
<img height="300" alt="Screenshot 2025-10-12 at 14 28 05" src="https://github.com/user-attachments/assets/6e8c917e-4b07-46c9-a083-b72f06a88562" />

v1ctorsales committed:
feat(calendar): redesign calendar cell layout for responsiveness and banner support

v1ctorsales commented:
@kKaskak Thx for the feedbacks!

I think due to to my screen resolution i couldnt see these edge cases. I tryied something more responsive this time.

I removed the 3 dots for every scenario, note that for really tiny screens (400px width or less it might broke).

However, for smartphone use it keeps working, see the image below:

smartphone
<img width="240" height="536" alt="image" src="https://github.com/user-attachments/assets/15381f44-ba1c-425b-838a-c3771802453f" />

desktop
<img width="1365" height="616" alt="image" src="https://github.com/user-attachments/assets/92003c64-2668-406e-a6e2-1dc6f88a1c27" />

kKaskak commented:
when the size of the screen requires you to show bottom sheet component you should not make the episode banners clickable 

<img height="300" alt="Screenshot 2025-10-13 at 10 19 04" src="https://github.com/user-attachments/assets/fa1619f4-20bb-4d46-bcab-8456db6c1015" />

kKaskak commented:
for some reason the banners are very widely spread out in some cases ?

<img width="169" height="302" alt="Screenshot 2025-10-13 at 10 20 49" src="https://github.com/user-attachments/assets/4b8f2144-4d90-4281-99b5-15efa98e67a7" />

v1ctorsales committed:
refactor(calendar): replace fixed width with max-width for better banner scaling

v1ctorsales committed:
fix(calendar): disable banner click in phone-portrait mode

v1ctorsales commented:
@kKaskak i commited both changes, but the view that you have here:
<img width="240" height="536" alt="image" src="https://github.com/user-attachments/assets/77aecec5-cfe0-4cc3-b9ef-6f7a83d9ef25" />

is only available using the browser using custom resolutions, if you change for native cellphone resolutions, the display for the calendar day and the banner should be in columns, like here:
<img width="240" height="536" alt="image" src="https://github.com/user-attachments/assets/3ca03d8f-5c6b-4781-a4ca-ca91148b393a" />

if you had that view in a non-native resolution, please lmk ;)

kKaskak commented:
please use custom rules such as `@phone-landscape`, `@phone-portrait` etc. you can find these in `screen-sizes.less`

ofc where possible

v1ctorsales committed:
style(responsive): add @phone-landscape media query

v1ctorsales commented:
@kKaskak i was already using @phone-portrait but i added @phone-landscape, i kept the other media queries for desktop

kKaskak commented:
> @kKaskak i was already using @phone-portrait but i added @phone-landscape, i kept the other media queries for desktop

seems better than before thank you 

however please refactor these cases 

1. when on desktop and there is enough space show the items in the blue rect instead to use the available space
 
<img height="300" alt="Screenshot 2025-10-13 at 16 53 06" src="https://github.com/user-attachments/assets/43768956-b695-4382-8ba8-9ad093cb5798" />

2. please make sure the active day indicator is a circle 

<img height="200" alt="Screenshot 2025-10-13 at 16 53 53" src="https://github.com/user-attachments/assets/a4dc22fe-3108-4059-aec8-2e13f2284d66" />

3. make sure when we show the bottom sheet we cannot click on any items as i mentioned before, clicking on an item should never have 2 outcomes on mobile (bad UX, you dont know what will happen since the cell is small)
so always when the bottom sheet will show disable item clicks 

<img  height="300" alt="Screenshot 2025-10-13 at 16 56 59" src="https://github.com/user-attachments/assets/b1b5d352-e3c3-4e2c-affc-4b9fb031c23b" />

4. on ipad there seems to be an inconsistency since once the items are too small and they do not fill space once too big ?

<img height="300" alt="Screenshot 2025-10-13 at 16 55 14" src="https://github.com/user-attachments/assets/ed805fc6-693e-441f-9581-87831279fa59" />
<img height="300" alt="Screenshot 2025-10-13 at 16 55 01" src="https://github.com/user-attachments/assets/deecec4d-6b0d-415a-a00c-90458abf23f9" />

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -134,56 +138,88 @@
-        .heading {
-            justify-content: center;
+@media only screen and (max-width: @minimum) and (orientation: portrait) {
+  .cell{
```
please follow project less code styles structure with indentations

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -134,56 +138,88 @@
+      display: flex;
+
+      .items{
+          .item{
```
please follow project less code styles structure with indentations

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -134,56 +138,88 @@
+          display: flex;
+          gap: 0.25rem;
+          width: auto; 
+          .day{
```
please follow project less code styles structure with indentations

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -134,56 +138,88 @@
+      flex-direction: row;
+      display: flex;
+
+      .items{
```
please follow project less code styles structure with indentations

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -64,7 +65,9 @@
             height: 100%;
             aspect-ratio: 2 / 3;
-            border-radius: var(--border-radius);
+            border-radius: calc(var(--border-radius) /2);
```
please follow project less code styles structure with `Space`

kKaskak commented on .vscode/settings.json:
please remove this file from the commit

v1ctorsales replied:
file removed!

v1ctorsales committed:
fix(calendar): adaptive display and style fixes

kKaskak commented:
hi looks better but you did not handle the play button cases that i mentioned before 
and also i proposed on mobile to keep the posters full cell size 
here are the problems i encountered now
- too small on mobile and not aligned 
-  the day label jumps ? liek when you have an item for a day and poster visible its in a diff place

<img  height="300" alt="Screenshot 2025-10-15 at 11 15 40" src="https://github.com/user-attachments/assets/f6e4b56b-510c-4083-a271-885163ec7fee" />
<img  height="300" alt="Screenshot 2025-10-15 at 11 15 13" src="https://github.com/user-attachments/assets/e99255ac-0bcc-4b7b-9e9c-a299457cd9e9" />

v1ctorsales committed:
fix: adapt items display for mobile landscape

v1ctorsales committed:
fix: align banners with day for small desktops

v1ctorsales commented:
@kKaskak, thanks for the feedback! I’ve adjusted the layout for mobile landscape from column to row, and fixed the 'unaligned' day text that would appear whenever a banner was displayed on that day.

Regarding the play button, it only appears in desktop view. When switching to mobile, the play button disappears, and there's just one outcome when the user clicks on the cell.

If you’re experiencing a different outcome, could you please share the resolution and zoom percentage you're using?

kKaskak commented:
> @kKaskak, thanks for the feedback! I’ve adjusted the layout for mobile landscape from column to row, and fixed the 'unaligned' day text that would appear whenever a banner was displayed on that day.
> 
> Regarding the play button, it only appears in desktop view. When switching to mobile, the play button disappears, and there's just one outcome when the user clicks on the cell.
> 
> If you’re experiencing a different outcome, could you please share the resolution and zoom percentage you're using?

hi again, good job with the PR, UI wise it's much better after your recent changes.

The case that is happening is on iPad Pro with `orientation: portrait`.

here is an example 
<img height="300" alt="Screenshot 2025-10-16 at 18 20 02" src="https://github.com/user-attachments/assets/7625ced1-63fb-4ab3-a612-6c2dbc61ed92" />

PS: i think that is the last think that needs to be done for this PR to be merged, congrats 🎊

v1ctorsales committed:
fix: point event none for tablets on portrait mode

v1ctorsales commented:
@kKaskak thanks for the feedback once again! i commited the change to adjust it

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -206,3 +206,13 @@
         }
     }
 }
+
```
```suggestion
@media screen and (max-width: @small) and (orientation: portrait) {
    .disable-cell-items();
}
```

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```diff
@@ -134,56 +137,82 @@
+            .item {
+                pointer-events: none;
+            }
         }
+    }
+}
```
```suggestion
@media only screen and (max-width: @minimum) {
    .disable-cell-items();
}
```

kKaskak commented on src/routes/Calendar/Table/Cell/Cell.less:
```suggestion

.disable-cell-items() {
    .cell {
        .items {
            .item {
                pointer-events: none;
            }
        }
    }
}

```

kKaskak commented:
you can reuse the less variable in other places as well

v1ctorsales committed:
refactor: reduce duplicated CSS using less variables

v1ctorsales commented:
@kKaskak done! i also create other variable for compacting items on mobile

kKaskak commented:
> @kKaskak done! i also create other variable for compacting items on mobile

Pls remove the min width on the last media query line 221 as well

v1ctorsales committed:
fix: remove unused mediaquery

v1ctorsales commented:
> > @kKaskak done! i also create other variable for compacting items on mobile
> 
> Pls remove the min width on the last media query line 221 as well

line 212 u mean? i removed the entire meadia query

kKaskak commented:
> > > @kKaskak done! i also create other variable for compacting items on mobile
> 
> > 
> 
> > Pls remove the min width on the last media query line 221 as well
> 
> 
> 
> line 212 u mean? i removed the entire meadia query

Yeah sorry I replied from my phone and didn't get the newest commit all good now
